sdk/go/pulumi/resource_test.go

```diff
@@ -548,16 +546,15 @@ func TestDependsOnInputs(t *testing.T) {
 			res := newTestRes(t, ctx, "res", opts)
 			assertHasDeps(t, ctx, depTracker, res, dep1, dep2)
 			return nil
-		}, WithMocks("project", "stack", &testMonitor{}))
+		}, WithMocks("project", "stack", &testMonitor{}), WrapResourceMonitorClient(depTracker.Wrap))
 		assert.NoError(t, err)
 	})
 
 	t.Run("dynamic", func(t *testing.T) {
 		t.Parallel()
 
+		depTracker := new(dependenciesTracker)
```

Comment:
nit but nearly everywhere else in the codebase we do `&dependenciesTracker{}`, and it's the same thing but we should probably stay consistent.